developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,420 @@
+        sample_sizes s8 s16 s24 s32
+        sample_containers c8 c16 c24 c32
+        channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1
+
+        # mod_cfg [PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS]
+        mod_cfg = [0, 0, 0, 0, 416, 914000, 48, 64, 0, 0, 0,
+                   1, 0, 0, 0, 416, 1321600, 192, 256, 0, 0, 0,
+                   2, 0, 0, 0, 416, 1786000, 192, 256, 0, 0, 0,
+                   3, 0, 0, 0, 416, 2333000, 48, 64, 0, 0, 0,
+                   4, 0, 0, 0, 416, 2910000, 192, 256, 0, 0, 0,
+                   5, 0, 0, 0, 416, 3441000, 192, 256, 0, 0, 0,
+                   6, 0, 0, 0, 416, 4265000, 192, 256, 0, 0, 0]
+
+
+Module entry definition
+***********************
+
+Modules Configuration
+=====================
+The module design configuration is placed in binamp file. The module examples
+binmaps are placed in:
+``<repo_dir>/FW/intel_common/module_binmaps``
+Module binmaps contains information about several module parameters e.g. module
+uuid, module version, input/output pins, number of channels, sample size,
+schedule capabilities. Ultimately, binmaps should be customized according to
+user needs.
+
+Module name
+-----------
+**Usage:**
+
+``module module_type module_name``
+
+**Example:**
+
+``module o AMPLI``
+
+*module_type* can take following values:
+
+- *o*   - optional module
+- *b*   - built-in module
+- *ba*  - built-in auto-init module
+- *d*   - module packaged into separate file
+- *da*  - auto-init module packaged into seperate file
+- *l*   - internal common sections module
+
+*module_name* is a string name o module. It does not have to be unique. Should
+not exceed 8 characters.
+
+Module UUID
+-----------
+**Usage:**
+
+``uuid uuid_number``
+
+**Example:**
+
+``uuid 8075F4F8-6214-4A61-8C08-884BE5D14FF8``
+
+*uuid_number* unique UUID number. See [UUID Wikipedia](https://en.wikipedia.org/wiki/Universally_unique_identifier)
+for a description.
+
+Module pretty name
+------------------
+**Usage:**
+``name pretty_name``
+
+**Example:**
+``name Aca module example``
+
+*pretty_name* does not have to be unique. It is not limited to 8 characters.
+
+Module version
+--------------
+
+**Usage:**
+  * version_major major_value
+  * version_minor minor_value
+  * version_hotfix hotfix_value
+  * version_build build_value
+
+**Example:**
+  * version_major 0x1
+  * version_minor 0x0
+  * version_hotfix 0x0
+  * version_build 0x0
+
+- *version_major* - shall be used only for vendor parameter structure change
+  (user decide how to use version parameter);
+- *version_minor* - shall be used only for internal changes (optimizations,
+  bug fix);
+- *version_hotfix* - shall be used during the bug fixing period;
+- *version_build* - shall be incremented with new build.
+
+Module affinity
+---------------
+**Usage:**
+``affinity_mask mask``
+
+**Example:**
+``affinity_mask MASTER_CORE_AFFINITY``
```

Comment:
The term MASTER is no longer used in SOF documentation. Please use MAIN or another approved alternative.